### PR TITLE
shift packages to point of use

### PR DESCRIPTION
**Pull request by deads2k**
This is nothing but moves and dead code removal.  The moves were done to move packages closer to point of use (lots of single call helpers) and continue isolating `pkg/oc` from the rest of origin.

@bparees the only notable exceptions is moving the code used to read some of your controller config down under your controllers package.  It was not used by any other code in the repo.

### CODE REVIEW

**Comment by deads2k** [comment]
@openshift/sig-master 
@soltysh @juanvallejo  since this mostly affects `pkg/oc`

**Comment by bparees** [comment]
build changes seem ok to me.

File: pkg/cmd/version/version.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ type VersionInfo interface {
 func NewCmdVersion(fullName string, versionInfo VersionInfo, out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "version",
-		Short: "Display client and server versions",
```
Comment (soltysh) [inline pkg/cmd/version/version.go]: What's wrong with the slightly longer name?

Comment (deads2k) [reply pkg/cmd/version/version.go]: > What's wrong with the slightly longer name?

It doesn't display server versions.

**Comment by deads2k** [comment]
`[WARNING] Expected `/tmp/openshift/test-cmd/etcd` to be mounted on `tmpfs` but found `xfs` instead.`

How annoying.

/retest

**Pull request merged**